src/plugins/interactive_setup/server/routes/views/index.ts

```diff
@@ -6,12 +6,9 @@
  * Side Public License, v 1.
  */
 
-import type { CoreSetup, CoreStart, Plugin } from 'src/core/server';
+import type { RouteDefinitionParams } from '../';
+import { defineViewStateRoutes } from './view_state';
 
-export class UserSetupPlugin implements Plugin {
-  public setup(core: CoreSetup) {}
-
-  public start(core: CoreStart) {}
-
-  public stop() {}
+export function defineViewRoutes(params: RouteDefinitionParams) {
```

Comment:
What are view routes?

Reply:
Clarified here https://github.com/elastic/kibana/pull/105222/files#r680721754. We have similar concept in Security plugin already - all routes that are supposed to support views (custom JS files, custom HTML files, view states etc.), not API routes

Reply:
Aren't all of these plugin's routes view routes using that definitions? They're not really direct proxies for ES APIs but serve specific functions to support the UI. 

I'm not sure if that distinctions is necessary or makes things clearer. I'd be happy to drop the concept in favour of a flat structure since we likely won't be defining more than 3 routes or so.